storage: unsketch TestStoreRangeSplitRaceUninitializedRHS

tbg opened the pull request:
The former version was using a TestingCommandFilter which caused replica
divergence (while claiming the opposite).

cc #9459.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9476)

<!-- Reviewable:end -->

petermattis commented on storage/client_split_test.go:
```diff
@@ -1326,25 +1326,42 @@ func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
+		// nonexistent right-hand-side, giving it a head start. This code looks
+		// fairly complicated since it wants to ensure that the two replicas
+		// don't diverge.
 		if trigger != nil && len(trigger.RightDesc.Replicas) == 2 && args.Hdr.Txn.Epoch == 0 {
```
I would have thought `Epoch` would be incremented on the retry. Is that not the case?

tschottdorf replied:
Yes, but if that goes down into this path it will deadlock the test (as it comes with a new command ID).
At the time the test was written there were also Raft reproposals, but we have RaftAppliedIndex now which should prevent those.

petermattis reviewed (approved):
LGTM

tbg committed:
storage: unsketch TestStoreRangeSplitRaceUninitializedRHS

The former version was using a TestingCommandFilter which caused replica
divergence (while claiming the opposite).
